src/NetAppFiles/NetAppFiles/Volume/GetNetAppFilesBackupStatus.cs

```diff
@@ -0,0 +1,148 @@
+            "Microsoft.NetApp/netAppAccounts/capacityPools/volumes",
+            nameof(ResourceGroupName),
+            nameof(AccountName),
+            nameof(PoolName))]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "The resource id of the ANF volume")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            ParameterSetName = ParentObjectParameterSet,
+            HelpMessage = "The pool object containing the volume to return")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesPool PoolObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The volume object to get backup status for")]
+        [ValidateNotNullOrEmpty]
+        public PSNetAppFilesVolume InputObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName == ResourceIdParameterSet)
+            {
+                var resourceIdentifier = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                var parentResources = resourceIdentifier.ParentResource.Split('/');
+                AccountName = parentResources[1];
+                PoolName = parentResources[3];
+                Name = resourceIdentifier.ResourceName;
+            }
+            else if (ParameterSetName == ObjectParameterSet)
+            {
+                ResourceGroupName = InputObject.ResourceGroupName;
+                var NameParts = InputObject.Name.Split('/');
+                AccountName = NameParts[0];
+                PoolName = NameParts[1];
+                Name = NameParts[2];
+            }
+            else if (ParameterSetName == ParentObjectParameterSet)
+            {
+                ResourceGroupName = PoolObject.ResourceGroupName;
+                var NameParts = PoolObject.Name.Split('/');
+                AccountName = NameParts[0];
+                PoolName = NameParts[1];
+            }
+
+            Management.NetApp.Models.Volume existingVolume = null;
+            try
+            {
+                existingVolume = AzureNetAppFilesManagementClient.Volumes.Get(ResourceGroupName, AccountName, PoolName, Name);
+            }
+            catch
+            {
+                existingVolume = null;
+            }
+            if (existingVolume == null)
+            {
+                throw new Exception(string.Format("A Volume with name '{0}' in resource group '{1}' does not exists. Please use New-AzNetAppFilesVolume to create a new Volume.", this.Name, this.ResourceGroupName));
```

Comment:
please use AzPsExceptions
https://github.com/Azure/azure-powershell-common/tree/master/src/Common/Exceptions

Reply:
@VeryEarly  Sure makes sense.  
Note the example code in the powershell repo should perhaps be updated to reflect this.  
https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/examples/update-cmdlet-example.md